src/sqlfluff/core/rules/std/L006.py

```diff
@@ -30,113 +31,187 @@ class Rule_L006(BaseRule):
-            if segment.is_code:
-                # This is code, what kind?
-                if segment.is_type("binary_operator", "comparison_operator"):
-                    # It's an operator, we can evaluate whitespace before it.
-                    anchor, fixes = _handle_previous_segments(
-                        memory["since_code"],
-                        anchor=segment,
-                        this_segment=segment,
-                        fixes=fixes,
+                    # And it's not a placeholder that itself ends with whitespace.
+                    # NOTE: This feels convoluted but handles the case of '-%}' modifiers.
+                    and not (
+                        prev_seg.is_meta
+                        and (
+                            prev_seg.source_str.endswith(" ")
+                            or prev_seg.source_str.endswith("\n")
+                        )
                     )
-                    if anchor:
-                        description = "Operators should be preceded by a space."
-                else:
-                    # It's not an operator, we can evaluate what happened after an
-                    # operator if that's the last code we saw.
-                    if memory["last_code"] and memory["last_code"].is_type(
-                        "binary_operator", "comparison_operator"
-                    ):
-                        # Evaluate whitespace AFTER the operator
-                        anchor, fixes = _handle_previous_segments(
-                            memory["since_code"],
-                            anchor=memory["last_code"],
-                            this_segment=segment,
-                            fixes=fixes,
+                ):
+                    self.logger.debug(
+                        "Missing Whitespace Before %r. Found %r instead.",
+                        before_anchor.raw,
+                        prev_seg.raw,
+                    )
+                    violations.append(
+                        LintResult(
+                            anchor=before_anchor,
+                            description="Missing whitespace before {0}".format(
+                                before_anchor.raw[:10]
+                            ),
+                            fixes=[
+                                LintFix(
+                                    "create",
+                                    # NB the anchor here is always in the parent and not anchor
+                                    anchor=sub_seg,
+                                    edit=self.make_whitespace(
+                                        raw=" ", pos_marker=sub_seg.pos_marker
+                                    ),
+                                )
+                            ],
                         )
-                        if anchor:
-                            description = "Operators should be followed by a space."
+                    )
+
+            if check_after:
```

Comment:
Nitpick: The `after` section of this code looks like a mirror image of the `_before` section. Would it be worthwhile to try and have one version of this code that can be run in both directions (forward and backward)? (I'm torn on this myself.)

Reply:
I don't know if we can have the same code, but I've separated out two slightly more generic methods to reuse a chunk of the logic. That should at least minimize issues from repeated code. What do you think of it now?